### DISCUSSION TITLE
How to load tabtab completion?

### DISCUSSION THREAD

**Discussion by muuvmuuv** [marlonrichert/zsh-snap#180] [Q&A] [answered]
I would like to enable tabtab completion for `pnpm` with znap, but I don't see a way of loading a custom zsh completion script.

- https://pnpm.io/completion
- https://github.com/pnpm/tabtab

**Comment by marlonrichert** [comment, marked as answer]
`pnpm install-completion zsh` creates two new files:
* `~/.config/tabtab/zsh/__tabtab.zsh` and
* `~/.config/tabtab/zsh/pnpm.zsh`.

It then goes on to modify `~/.zshrc` (and if it doesn't exist, first creates it).

_**This**_ (modifying a user's dotfiles) _**is a big NO-NO in Unix land!**_ You should [file a bug report with `pnpm`](https://github.com/pnpm/pnpm/issues/new?assignees=&labels=type%3A+bug&template=bug-report.md&title=) about this. Plus, if your `$ZDOTDIR` is set to something else than `$HOME`, this won't even enable `pnpm` completion (because then, `$ZDOTDIR/.zshrc` is read at startup, instead of `~/.zshrc`.

What `pnpm` really should be doing is to install its completion function to `/usr/local/share/zsh/site-functions`, [as I described here on Stack Overflow](https://stackoverflow.com/a/67704746/13832769). You can include that link your bug report. If they do it the way I described, then there is no need to modify the user's dotfiles, because the completion function will get picked up by `compinit` automatically.

You can discard `~/.config/tabtab/zsh/__tabtab.zsh` and the changes to `~/.zshrc`. Instead, just add 
```zsh
source $XDG_CONFIG_HOME/tabtab/zsh/pnpm.zsh
```
to your `.zshrc` file, _after_ sourcing Znap.

In short: The way `pnpm` installs completions is incorrect. They really need to change it and there's no way you can manage this with Znap.

**Reply by muuvmuuv** [reply]
Many thanks for the detailed explanation! This makes a lot more sense. I hate it when installer modified any of my conf, so yes, reasonable. 

Once I get time again for my Zsh stuff I will fill an issue on pnpm. But I guess tabtab is doing it wrong from beginning.
